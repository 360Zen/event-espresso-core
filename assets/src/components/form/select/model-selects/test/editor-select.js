import EditorSelect from '../editor-select';
import { shallow } from 'enzyme';

describe( '<EditorSelect />', () => {
	it( 'matches snapshot with no props provided', () => {
		const wrapper = shallow(
			<EditorSelect />
		);
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'matches snapshot with all props provided but no child component', () => {
		const wrapper = shallow(
			<EditorSelect
				label={ 'not a default label' }
				id={ 'i-am-unique' }
				className={ 'not-so-unique' }
				help={ 'try not to fall down stairs!' }
			/>
		);
		expect( wrapper ).toMatchSnapshot();
	} );
	it( 'matches snapshot with only child component provided', () => {
		const wrapper = shallow(
			<EditorSelect>
				<p>{ 'child component' }</p>
			</EditorSelect>
		);
		expect( wrapper ).toMatchSnapshot();
	} );
} );

// location: assets/src/components/form/select/model-selects/test/editor-select.js
