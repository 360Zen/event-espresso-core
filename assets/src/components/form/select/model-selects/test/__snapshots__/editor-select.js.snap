// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`<EditorSelect /> matches snapshot with no props provided 1`] = `
<EditorSelect
  instanceId={0}
/>
`;

exports[ `<EditorSelect /> matches snapshot with all props provided but no child component 1` ] = `
<EditorSelect
  className="not-so-unique"
  help="try not to fall down stairs!"
  id="i-am-unique"
  instanceId={1}
  label="not a default label"
/>
`;

exports[ `<EditorSelect /> matches snapshot with only child component provided 1` ] = `
<EditorSelect
  instanceId={2}
>
  <p>
    child component
  </p>
</EditorSelect>
`;
